core/src/validator.rs

```diff
@@ -1388,7 +1388,7 @@ fn report_target_features() {
         }
     );
 
-    #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+    #[cfg(all(any(target_arch = "x86", target_arch = "x86_64"), not(target_os = "macos")))]
```

Comment:
macos m1 claims to be `target_arch = "x86"` or `target_arch = "x86_64"`, that seems terrible

Reply:
But is that just for the interpreted? There is the mode on m1 to execute x86 code, so it has to report that.

Reply:
wouldn't this feature check affect the x86 macs which have AVX?

Reply:
To get this going on an M1, you have to install all of your stuff with Rosetta 2, a tool that allows you to run x86_64 software. This amounts to using a version of rust/cargo etc. that targets `x86_64` instead of `aarch64`. Which produces this extremely confusing feature flag

Without Rosetta, rust would target `aarch64` – I tried to do this initially, and gave up after a day as there were several more blockers upstream

Reply:
Ah, got it.  Would you mind sneaking in a comment in the code then here about why we're excluding macos.  Otherwise somebody with an intel mbp (everybody on the core team) is likely to remove this line later inadvertently

Reply:
@sakridge yes it would and is a clear downside of including this feature check. If you could think of a better way to get around this, I'm happy to implement it and be your Mac M1 guinea pig guy

@mvines yes of course! Was planning on this and appreciate the reminder